misc/toml-parser/src/test/java/org/ballerinalang/toml/parser/ManifestProcessorTest.java

```diff
@@ -0,0 +1,260 @@
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[package] \n " +
+                "repository = \"https://github.com/ballerinalang/ballerina\"");
+        Assert.assertEquals(manifest.getRepositoryURL(), "\"https://github.com/ballerinalang/ballerina\"");
+    }
+
+    @Test(description = "Version in non-package section has no effect")
+    public void testVersionNeg() throws IOException {
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[patch] \n version = \"v1\"");
+        Assert.assertNotEquals(manifest.getVersion(), "\"v1\"");
+    }
+
+    @Test(description = "Location in package section has no effect")
+    public void testLocationNeg() throws IOException {
+        ManifestProcessor.parseTomlContentFromString("[package] \n location = \"local\"");
+        Assert.assertNotEquals(null, "\"local\"");
+    }
+
+    @Test(description = "Readme file path in package section has an effect")
+    public void testReadmeFilePath() throws IOException {
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[package] \n " +
+                "readme = \"https://github.com/ballerinalang/composer/blob/master/README.md\"");
+        Assert.assertEquals(manifest.getReadmeFilePath(), "\"https://github.com/ballerinalang/composer/blob" +
+                "/master/README.md\"");
+    }
+
+    @Test(description = "Keywords in package section has an effect")
+    public void testKeywords() throws IOException {
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[package] \n" +
+                "keywords=[\"ballerina\",\"security\",\"security\"]");
+        Assert.assertEquals(manifest.getKeywords().get(0), "\"ballerina\"");
+        Assert.assertEquals(manifest.getKeywords().get(1), "\"security\"");
+        Assert.assertEquals(manifest.getKeywords().size(), 3);
+    }
+
+    @Test(description = "Description in package section has an effect")
+    public void testLicenseDescription() throws IOException {
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[package] \n " +
+                "license = \"MIT OR Apache-2.0\"");
+        Assert.assertEquals(manifest.getLicense(), "\"MIT OR Apache-2.0\"");
+    }
+
+    @Test(description = "One dependancy added to the dependancies section has an effect")
+    public void testSingleDependancies() throws IOException {
+        Manifest manifest = ManifestProcessor.parseTomlContentFromString("[dependencies] \n " +
+                "string-utils = {location = \"src/string-utils\", version = \"0.15\"} \n");
+        Assert.assertEquals(manifest.getDependencies().get(0).getPackageName(), "string-utils");
+        Assert.assertEquals(manifest.getDependencies().get(0).getVersion(), "\"0.15\"");
+        Assert.assertEquals(manifest.getDependencies().get(0).getLocation(), "\"src/string-utils\"");
+    }
+
+    @Test(description = "Empty dependancy added to the dependancies section has no effect")
```

Comment:
Typo - should be dependency. check the whole file

Reply:
Removed all the custom listeners from the toml-parser